src/main/java/org/launchcode/liftoffproject/models/Child.java

```diff
@@ -31,23 +32,36 @@ public class Child extends AbstractEntity{
 
     private String avatar;
 
+    public Child() {
+    }
 
-    public Child(){}
+    //public Child(String firstName, String lastName, Parent parent, ChildUser newChildUser){}
 
-    public Child(String firstName, String lastName, Parent parent) {
-        this.firstName = firstName;
-        this.lastName = lastName;
-        this.parent = parent;
-    }
+//    public Child(String firstName, String lastName, Parent parent, ChildUser newChildUser) {
+//        super();
+//        this.firstName = firstName;
+//        this.lastName = lastName;
+//        this.parent = parent;
+//        this.userAccount = newChildUser;
+//    }
 
-    public Child(String firstName, String lastName, Parent parent, ChildUser childUser) {
+
+//     added constructor for reward
+    public Child(String firstName, String lastName, Parent parent, ChildUser userAccount) {
+
+        if (parent != null) {
+            Integer id = parent.getId();
+            if (id == null) {
+                throw new IllegalArgumentException("Parent ID cannot be null");
+            }
+        } else {
+            throw new IllegalArgumentException("Parent cannot be null");
+        }
```

Comment:
Don't get why we need to have a constructor for the reward; the Child object is already created in the database whenever we are trying to award points.